core/src/main/java/org/apache/druid/indexer/Jobby.java

```diff
@@ -30,6 +30,17 @@
 {
   boolean run();
 
+  /**
+   *
+   * @return A string represtenting the jobId of the actual MR job.
+   * Run method is now divided into two parts. The first one being submitAndGetHadoopJobId which just submits the job and returns the job ID
+   * Run then monitors this job for completion
+   */
+  default String submitAndGetHadoopJobId()
```

Comment:
`DeterminePartitionsJob` should implement this method and use `JobHelper`.

Reply:
Sorry, I'm not sure what you mean. Would you tell me more details?

Reply:
So currently the run() issues two MR jobs
1. [determine_partitions_groupby](https://github.com/apache/incubator-druid/blob/master/indexing-hadoop/src/main/java/org/apache/druid/indexer/DeterminePartitionsJob.java#L161)
2. [determine_partitions_dimselection](https://github.com/apache/incubator-druid/blob/master/indexing-hadoop/src/main/java/org/apache/druid/indexer/DeterminePartitionsJob.java#L214)

I need these job to return their jobIds first and then check for their status, hence planning to split run to two parts

Basically, case when job is an instance of DetermineJobPartition [JobHelper.runSingleJob()](https://github.com/ankit0811/incubator-druid/blob/feature-killMRJob/indexing-hadoop/src/main/java/org/apache/druid/indexer/JobHelper.java#L353) will look like this

``` java
 if (job instanceof DeterminePartitionsJob) {
        String hadoopJobId = ((DeterminePartitionsJob) job).submitAndGetHadoopJobIdForDeterminePartitionsGroupBy();
((DeterminePartitionsJob) job).RunDeterminePartitionsGroupBy();
```

the above will take care of the [determine_partitions_groupby job](https://github.com/apache/incubator-druid/blob/master/indexing-hadoop/src/main/java/org/apache/druid/indexer/DeterminePartitionsJob.java#L161)
followed by submitAndGetHadoopJobId and run() (the normal code flow) which will take care of [determine_partitions_dimselection job](https://github.com/apache/incubator-druid/blob/master/indexing-hadoop/src/main/java/org/apache/druid/indexer/DeterminePartitionsJob.java#L214)

Reply:
Thanks, but I think it's too specific for an implementation of `DeterminePartitionsJob` which can't handle any other custom `Jobby` implementations running two or more Hadoop jobs. 

I think maybe it's not a good idea to add `submitAndGetHadoopJobId` to `Jobby` because a single `Jobby` can run 0, 1, or more Hadoop jobs. How about removing `submitAndGetHadoopJobId` from `Jobby` but adding `writeHadoopJobId` to `JobHelper`? Every `Jobby` running one or more Hadoop jobs should use this method.

Reply:
Yes
That simplifies things 👍 
Thanks. Will make the changes